analyses/cnv-chrom-plot/cn_status_heatmap.Rmd

```diff
@@ -0,0 +1,452 @@
+  df = data.frame(chrs),
+  col = list(chrs = chr_colors),
+  name = "",
+  show_legend = FALSE,
+  show_annotation_name = FALSE,
+  mark = chr_text # Put the text in
+)
+```
+
+### Make row annotation object
+
+Make histology labeling `HeatmapAnnotation` object.
+This annotation object strategy was originally from [chromosomal-instability](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/b5a33838d1e9bd7e7913a89201ec26125c16c94c/analyses/chromosomal-instability/02a-plot-chr-instability-heatmaps.Rmd#L73).
+
+```{r}
+# Get the histologies for the samples in this set and order them by histology
+histologies <-
+  data.frame(Kids_First_Biospecimen_ID = bin_calls_df$biospecimen_id) %>%
+  dplyr::inner_join(metadata %>%
+    dplyr::select(Kids_First_Biospecimen_ID, short_histology, sample_color)) %>%
+  dplyr::mutate(short_histology = factor(short_histology)) %>%
+  dplyr::arrange(short_histology) %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID")
+
+# Make color key specific to these samples
+histologies_color_key_filtered <- unique(histologies$sample_color)
+names(histologies_color_key_filtered) <- unique(histologies$short_histology)
+
+# Drop this column so ComplexHeatmap isn't tempted to plot it
+histologies <- dplyr::select(histologies, -sample_color)
+
+# Get coordinate start positions
+hist_pos <- match(names(histologies_color_key_filtered), histologies$short_histology)
+
+# Get mid points of chromosome labels
+mid_points <- ceiling((c(hist_pos[2:length(hist_pos)], length(histologies$short_histology)) + hist_pos[1:length(hist_pos)]) / 2)
+```
+
+```{r}
+# Make text labels for chromosome text
+hist_text <- ComplexHeatmap::anno_mark(
+  at = mid_points,
+  labels = levels(histologies$short_histology),
+  which = "row",
+  side = "right",
+  labels_gp = grid::gpar(cex = 0.65),
+  link_width = grid::unit(15, "mm")
+)
+
+# Create the Heatmap annotation object
+hist_annot <- ComplexHeatmap::HeatmapAnnotation(
+  df = data.frame(histologies),
+  col = list(short_histology = histologies_color_key_filtered),
+  which = "row",
+  show_annotation_name = FALSE,
+  show_legend = FALSE,
+  mark = hist_text # Put the text in
+)
+```
+
+Format `bin_calls_df` as a matrix with rownames for `ComplexHeatmap` to use. 
+
+```{r}
+bin_calls_mat <- bin_calls_df %>%
+  tibble::column_to_rownames("biospecimen_id") %>%
+  as.matrix()
+
+# Ensure that this matrix is in the same order as the annotation
+bin_calls_mat <- bin_calls_mat[rownames(histologies), ]
+
+# Double check its in thte same order
+all.equal(rownames(bin_calls_mat), rownames(histologies))
+```
+
+## Assemble CN status heatmap
+
+```{r}
+# Plot on a heatmap
+heatmap <- ComplexHeatmap::Heatmap(
+  bin_calls_mat,
+  name = "CN status",
+  col = color_key,
+  cluster_columns = FALSE,
+  cluster_rows = FALSE,
+  show_column_names = FALSE,
+  show_row_names = FALSE,
+  bottom_annotation = chr_annot,
+  right_annotation = hist_annot,
+  heatmap_legend_param = list(nrow = 1)
```

Comment:
The heatmap itself is drawn is rasterized, which is mostly fine, but may be a bit lower resolution than ideal. Can we see what happens if we bump it up, both to file size and image quality?  You might even go higher than this, depending on results. It is a bit hard to interpret the docs https://jokergoo.github.io/ComplexHeatmap-reference/book/a-single-heatmap.html#heatmap-as-raster-image, but if the general setting is equivalent to 72dpi, we might want to go as far as 4 here.
```suggestion
  heatmap_legend_param = list(nrow = 1),
  raster_quality = 2
```